smartling-api-commons/src/main/java/com/smartling/api/v2/client/AbstractApiFactory.java

```diff
@@ -18,16 +19,35 @@
  *
  * @param <T> the API interface type to proxy
  */
+@Slf4j
 public abstract class AbstractApiFactory<T> implements ApiFactory<T>
 {
+    static ClientFactory defaultClientFactory()
```

Comment:
so its in public class we try to load internal implementation first... seems unusual, but I guess its the best we can do here, right?

Reply:
Ok, let's use the other approach we discussed with making sure all API factories export both constructors. Will update PR, thanks!